### DISCUSSION TITLE
nbt in c#

### DISCUSSION THREAD

**Discussion by Vexea1** [HypixelDev/PublicAPI#427] [Q&A]
Hello,

I've got the bytes from the nbt data in the inv_armor property. But how do I convert this into a json object or array of bytes that represents a json format in c#. Please help!

            byte[] buffer = Convert.FromBase64String(value);
            byte[] decompressed;

            using (var inputStream = new MemoryStream(buffer))
            {
                using var outputStream = new MemoryStream();

                using (var gzip = new GZipStream(inputStream, CompressionMode.Decompress, leaveOpen: true))
                {
                    gzip.CopyTo(outputStream);
                }

                decompressed = outputStream.ToArray();
                byte[] json = new byte[];//Convert from nbt bytes to json bytes...
                JsonConvert.DeserializeObject<T>(Encoding.UTF8.GetString(json));
            }

**Comment by saiintbrisson** [comment]
I'd suggest using an existing NBT library for C#, like [this one](https://github.com/cyotek/Cyotek.Data.Nbt), doing it by hand will take some time. If you want to dig into the specification and implement it yourself, [this](https://wiki.vg/NBT#Specification) should be helpful.
